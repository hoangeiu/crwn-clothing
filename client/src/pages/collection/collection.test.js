import React from "react";
import { shallow } from "enzyme";

import { CollectionPage } from "./collection.component";
import CollectionItem from "../../components/collection-item/collection-item.component";

let wrapper;
let mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }];
beforeEach(() => {
  const mockCollection = {
    title: "hats",
    items: mockItems,
  };

  wrapper = shallow(<CollectionPage collection={mockCollection} />);
});

it("should CollectionPage render without error", () => {
  expect(wrapper).toMatchSnapshot();
});

it("should render an equal number of CollectionItem as collection prop", () => {
  expect(wrapper.find(CollectionItem).length).toEqual(mockItems.length);
});
