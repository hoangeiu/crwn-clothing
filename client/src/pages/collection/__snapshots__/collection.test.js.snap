// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should CollectionPage render without error 1`] = `
<div
  className="collection-page"
>
  <h2
    className="title"
  >
    hats
  </h2>
  <div
    className="items"
  >
    <Connect(CollectionItem)
      item={
        Object {
          "id": 1,
        }
      }
      key="1"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 2,
        }
      }
      key="2"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 3,
        }
      }
      key="3"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 4,
        }
      }
      key="4"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 5,
        }
      }
      key="5"
    />
  </div>
</div>
`;
